src/pybind/mgr/prometheus/module.py

```diff
@@ -258,22 +269,25 @@ def get_pg_status(self):
             if state not in reported_states:
                 self.metrics[path].set(0)
 
-    def get_metadata(self):
+    def get_metadata_and_osd_status(self):
         osd_map = self.get('osd_map')
         osd_dev = self.get('osd_map_crush')['devices']
         for osd in osd_map['osds']:
             id_ = osd['osd']
-            p_addr = osd['public_addr']
-            c_addr = osd['cluster_addr']
-            w = osd['weight']
+            p_addr = osd['public_addr'].split(':')[0]
+            c_addr = osd['cluster_addr'].split(':')[0]
             dev_class = next((osd for osd in osd_dev if osd['id'] == id_))
             self.metrics['osd_metadata'].set(0, (
                 c_addr,
                 dev_class['class'],
                 id_,
-                p_addr,
-                w
+                p_addr
             ))
+            for state in OSD_STATUS:
+                status = osd[state]
+                self.metrics['osd_{}'.format(state)].set(
+                    status,
+                    (id_,))
```

Comment:
This is setting the ceph_daemon label to just a number: it should be osd.0 or similar

Reply:
ah good catch...fix pushed, rebase coming up